go/batch-submitter/drivers/sequencer/encoding.go

```diff
@@ -191,7 +290,7 @@ func (p *AppendSequencerBatchParams) Read(r io.Reader) error {
 }
 
 // writeUint64 writes a the bottom `n` bytes of `val` to `w`.
-func writeUint64(w *bytes.Buffer, val uint64, n uint) {
+func writeUint64(w io.Writer, val uint64, n uint) {
```

Comment:
If we're changing the interface to use an `io.Writer` rather than `bytes.Buffer`, we should return an error here since writes are no longer guaranteed to never fail.

Reply:
Noting that this change is due to a `zlib.Writer` being passed to the function

Reply:
yeah i think that's a good idea. even though the underlying writes to the buffer can't fail there are now possible errors that are independent of the underlying writer, e.g. https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/compress/zlib/writer.go;l=132

Reply:
This has been implemented